cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -249,7 +259,7 @@ DI BinT pdf_to_cdf(BinT* shared_histogram, IdxT nbins)
   // variable to accumulate aggregate of sumscans of previous iterations
   BinT total_aggregate = BinT();
 
-  for (IdxT tix = threadIdx.x; tix < raft::ceildiv(nbins, TPB) * TPB; tix += blockDim.x) {
+  for (std::size_t tix = threadIdx.x; tix < raft::ceildiv(nbins, TPB) * TPB; tix += blockDim.x) {
```

Comment:
`threadIdx.x` can only be integer (signed 4 bytes), making it `size_t` is not going to be useful.